buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/CreateRuleStubsTask.kt

```diff
@@ -0,0 +1,238 @@
+        val checkListEntryToAdd = "    $checkClassName::class.java,"
+        for (i in 0 until read.size - 1) {
+            val currentLine = read[i]
+            val nextLine = read[i + 1]
+            when (fileSegment) {
+                0 -> { // Header before imports
+                    toWrite.add(currentLine)
+                    if (nextLine.startsWith("import ")) fileSegment++
+                }
+                1 -> { // imports
+                    if (currentLine == importStatementToAdd) {
+                        // In case the import is already in the list don't add it a second time
+                        fileSegment++
+                    } else if (currentLine > importStatementToAdd || !currentLine.startsWith("import ")) {
+                        // Add import to the list, in alphabetical ordering
+                        toWrite.add(importStatementToAdd)
+                        fileSegment++
+                    }
+                    toWrite.add(currentLine)
+                }
+                2 -> { // in between imports & check list
+                    toWrite.add(currentLine)
+                    if (nextLine.contains("::class.java,")) fileSegment++
+                }
+                3 -> { // check list
+                    if (currentLine == checkListEntryToAdd) {
+                        // In case the check is already in the list don't add it a second time
+                        fileSegment++
+                    } else if (currentLine > checkListEntryToAdd || !currentLine.contains("::class.java")) {
+                        // Add check to the list, in alphabetical ordering
+                        toWrite.add(checkListEntryToAdd)
+                        fileSegment++
+                    }
+                    toWrite.add(currentLine)
+                }
+                else -> toWrite.add(currentLine)
+            }
+        }
+        toWrite.add(read.last())
+
+        checkListFile.writeLines(toWrite)
+
+        return fileSegment == 4
+    }
+
+    private fun downloadMetadata(): Boolean {
+        project.repositories {
+            maven {
+                url = project.uri("https://repox.jfrog.io/repox/sonarsource-private-releases")
+                authentication {
+                    credentials {
+                        val artifactoryUsername: String by project
+                        val artifactoryPassword: String by project
+                        username = artifactoryUsername
+                        password = artifactoryPassword
+                    }
+                }
+            }
+        }
+
+        val ruleApi = project.configurations.create("ruleApi")
+        project.dependencies {
+            ruleApi("com.sonarsource.rule-api:rule-api:2.0.0.1885")
+        }
+
+        val exec = project.javaexec {
+            classpath = project.files(ruleApi.resolve())
+            args = listOf("generate", "-rule", ruleKey)
```

Comment:
as far as I remember ruleApi requires env variable GITHUB_TOKEN to be set. Do you think it will be valuable to check if the variable is set before executing?